src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,121 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int MAX_CAPACITY = 10;
+    private Object[] array = new Object[MAX_CAPACITY];
+    private int size;
+
+    public void grow() {
+        Object[] newArray = new Object[size + size / 2];
+        for (int i = 0; i < size; i++) {
+            newArray[i] = array[i];
+        }
+        array = newArray;
+    }
+
     @Override
     public void add(T value) {
-
+        array[size] = value;
+        size++;
+        if (size == array.length) {
+            grow();
+        }
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index < 0 || index > size) {
+            throw new ArrayListIndexOutOfBoundsException(
+            "Array list index out of bounds. " + index);
+        }
+        if (size == array.length) {
+            grow();
+        }
+        for (int i = size; i > index; i--) {
+            array[i] = array[i - 1];
+        }
+        array[index] = value;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
+        if ((size + list.size()) > array.length) {
+            grow();
+        }
+        for (int i = 0; i < list.size(); i++) {
+            if (array.length == size) {
+                grow();
+            }
+            array[size] = list.get(i);
+            size++;
 
+        }
```

Comment:
just reuse add(T value) method in loop. It will determine required amount of resizes